CMR/python/README.md

```diff
@@ -0,0 +1,54 @@
+# CMR Python Wrapper
+
+A wrapper for [github.com/nasa/Common-Metadata-Repository][git_cmr] in python
+
+## Assumptions
+
+1. Python 3.2 or better.
+2. Can access [cmr.earthdata.nasa.gov][cmr]
+3. Optional: an account on [urs.earthdata.nasa.gov][edl]
+4. Third party libraries are not to be used unless absolutely necessary (none as of now)
+
+## Styles
+
+1. Coding Styles are defined as [PEP 8][pep8].
+2. This project uses the [editorconfig.org][econfig] file .editorconfig 
+3. Use: `pylint *` to check code
+
+## Usage
+
+### Getting a token
+
+There are three ways to store a password for use in scripts. Plain text in the script (not recommended), in a file, but still clear text, or if you are on a mac or compatible system, you can use the password manager which on Mac OS X will come from Keychain. These three methods are selected by passing in a lambda function to the token() function
+
+    import cmr.auth.token as t
+    edl_token = t.token(user_id)
+    edl_token = t.token(user_id, t.password("clear_text"))
+    edl_token = t.token(user_id, t.password_file, {"password_file":"~/.passwd"})
+    edl_token = t.token(user_id, t.password_manager, {})
+
+* user_id is the Earthdata login name
+* `password_lambda` is a function for returning passwords and defaults to `password_file`.
+    * `password(text)` = creates a lambda using a hard coded password.
+    * `password_file` = uses the contents of `~/.cmr_password`. File can be overwritten by setting `password.path`
```

Comment:
do we want to use the home directory for this explicitly? do we want to allow for configuration?